ext/sockets/sockets.stub.php

```diff
@@ -0,0 +1,176 @@
+
+/** @param resource $socket */
+function socket_set_nonblock($socket): bool {}
+
+/** @param resource $socket */
+function socket_set_block($socket): bool {}
+
+/** @param resource $socket */
+function socket_listen($socket, int $backlog = 0): bool {}
+
+/** @param resource $socket */
+function socket_close($socket): void {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_write($socket, string $buf, int $length = 0) {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return string|false
+ */
+function socket_read($socket, int $length, int $type = PHP_BINARY_READ) {}
+
+/** @param resource $socket */
+function socket_getsockname($socket, &$addr, &$port = UNKNOWN): bool {}
+
+/** @param resource $socket */
+function socket_getpeername($socket, &$addr, &$port): bool {}
+
+/** @return resource|false */
+function socket_create(int $domain, int $type, int $protocol) {}
+
+/** @param resource $socket */
+function socket_connect($socket, string $addr, int $port = 0): bool {}
+
+function socket_strerror(int $errno): string {}
+
+/** @param resource $socket */
+function socket_bind($socket, string $addr, int $port = 0): bool {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_recv($socket, &$buf, int $len, int $flags) {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_send($socket, string $buf, int $len, int $flags) {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_recvfrom($socket, &$buf, int $len, int $flags, &$name, &$port) {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_sendto($socket, string $buf, int $len, int $flags, string $addr, int $port = 0) {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_get_option($socket, int $level, int $optname) {}
+
+/** @param resource $socket */
+function socket_set_option($socket, int $level, int $optname, $optval): bool {}
+
+#ifdef HAVE_SOCKETPAIR
+function socket_create_pair(int $domain, int $type, int $protocol, &$fd): bool {}
+#endif
+
+#ifdef HAVE_SHUTDOWN
+/** @param resource $socket */
+function socket_shutdown($socket, int $how = 2) {}
+#endif
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_last_error($socket) {}
+
+/** @param resource $socket */
+function socket_clear_error($socket = UNKNOWN) {}
+
+/**
+ * @param resource $stream
+ * 
+ * @return resource|false
+ */
+function socket_import_stream($stream) {}
+
+/** @param resource $socket */
+function socket_export_stream($socket) {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_sendmsg($socket, array $msghdr, int $flags = 0) {}
+
+/**
+ * @param resource $socket
+ * 
+ * @return int|false
+ */
+function socket_recvmsg($socket, array &$msghdr, int $flags = 0) {}
+
+function socket_cmsg_space(int $level, int $type, int $n = 0): ?int {}
+
+/** @return resource|false */
```

Comment:
I think this function returns `array`, not `resource`, so:
```suggestion
/** @return array|false */
```